TombEngine/Objects/Effects/EmberEmitter.cpp

```diff
@@ -0,0 +1,157 @@
+#include "framework.h"
+#include "Objects/Effects/EmberEmitter.h"
+
+#include "Game/effects/effects.h"
+#include "Game/Setup.h"
+#include "Math/Math.h"
+
+using namespace TEN::Math;
+
+namespace TEN::Effects::EmberEmitter
+{
+	void ControlEmberEmitter(short itemNumber)
+	{
+		auto& item = g_Level.Items[itemNumber];
+
+		if (!TriggerActive(&item))
+			return;
+
+		unsigned char r = 0;
+		unsigned char g = 0;
+		unsigned char b = 0;
+
+		if (item.Model.Color.x == item.Model.Color.y &&
```

Comment:
I don't fully understand this if-then condition. Does it mean that if object tint is grayscale, no brightness variation in color will occur?

Reply:
Yes, with the gray color, the more the color goes into black, the more it would look like rainbow confetti because of the blend mode. Since I want to keep the rainbow confetti as an Easter egg, and because I also want to keep the color gray, I have excluded the brightness variation with a gray value where r, g and b are the same.

Reply:
What if you just variate all values (RGB) using same random number? Then they will remain grayscale but still will variate in brightness. In fact, I think similar approach must be used for any colour, this will prevent color drift (e.g. red particles become slightly magenta and so on).

Reply:
ah ok, I will try that

Reply:
yes, this fixed the issue